controller/api/destination/endpoint_translator.go

```diff
@@ -339,19 +343,69 @@ func (et *endpointTranslator) filterAddresses() watcher.AddressSet {
 func (et *endpointTranslator) diffEndpoints(filtered watcher.AddressSet) (watcher.AddressSet, watcher.AddressSet) {
 	add := make(map[watcher.ID]watcher.Address)
 	remove := make(map[watcher.ID]watcher.Address)
```

Comment:
This updated diffing logic is complex and difficult to think through all the cases and convince ourselves that we didn't miss any.  I think this could be simplified by breaking it into two discrete steps:
1. A `func selectAddressFamily(addresses watcher.AddressSet) watcher.AddressSet` which filters the address set based on address family:
  * if enable-ipv6 is disabled, drop all ipv6 addresses
  * if the address set contains both versions of an address, drop the ipv4 one
2. This will result in the filtered address set being the final desired state, which means we can just let `diffEndpoints` do it's thing and diff against the snapshot without being specifically aware of address family. it should generate the diff necessary to bring the filtered snapshot to the desired state.

Reply:
The expanded diffing logic was also trying to avoid generating spurious messages, like a Remove that didn't correspond to any previous Add, _but I may be over complicating things. Let me experiment with your proposed approach and see how it goes. I should also include the additional unit tests first to be more confident about these changes.